packages/flutter_tools/lib/src/context_runner.dart

```diff
@@ -172,6 +174,15 @@ Future<T> runInContext<T>(
         xcode: globals.xcode,
         platform: globals.platform,
       ),
+      IProxy: () => IProxy(
```

Comment:
@jonahwilliams another option is to remove this from the high level context, and instead make it required to initialize `XCDevice` since that's the only place it's used.
https://github.com/flutter/flutter/blob/859fce9010895ce5bcdb913a8373b5b1a194a1fb/packages/flutter_tools/lib/src/context_runner.dart#L250-L257

would become:
```dart
 XCDevice: () => XCDevice( 
   processManager: globals.processManager, 
   logger: globals.logger, 
   artifacts: globals.artifacts, 
   cache: globals.cache, 
   platform: globals.platform, 
   xcode: globals.xcode, 
   iproxy: IProxy(
     iproxyPath: globals.artifacts.getArtifactPath(
       Artifact.iproxy,
       platform: TargetPlatform.ios,
     ),
     logger: globals.logger,
     processManager: globals.processManager,
     dyLdLibEntry: globals.cache.dyLdLibEntry,
   ), 
```

This would make
https://github.com/flutter/flutter/pull/61762/files#diff-ebc648b2067d73f2af30539f5d3a59faR503
become:
```dart
iProxy: _iproxy,
```

Reply:
IMO Leaving/Removing from the context injection isn't really that important. As long as we're not introducing more globals into classes that have already been cleaned up, I have a separate plan to remove the context w/o breaking g3

Reply:
`XCDevice` is close to being cleaned up, I already did this exercise with https://github.com/flutter/flutter/pull/53203 (which was further pushed down in https://github.com/flutter/flutter/pull/5330,  we can't do in this case since g3 wants to override). I convinced myself, I'm going to do this pattern from the beginning.